src/Bicep.Core/TypeSystem/Providers/IProviderSpecificationSyntax.cs

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Bicep.Core.Syntax.Providers;
+
+public interface IProviderSpecificationSyntax : ISymbolNameSource
```

Comment:
Should we remove the `Syntax` suffix since this is not a syntax node type?

Reply:
+1, and same comment on the classes extending this.